sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/pipeline_component.py

```diff
@@ -334,6 +336,21 @@ def _get_anonymous_hash(self) -> str:
         # command component), so we just use rest object to generate hash for pipeline component,
         # which doesn't have reuse issue.
         component_interface_dict = self._to_rest_object().properties.component_spec
+        # Hash local inputs in pipeline component jobs
+        for job_name, job in self.jobs.items():
+            if getattr(job, "inputs", None):
+                for input_name, input_value in job.inputs.items():
+                    try:
+                        if (
+                            isinstance(input_value._data, Input)
+                            and input_value.path
+                            and os.path.exists(input_value.path)
```

Comment:
shall do check like path not starts with azureml here ?

Reply:
There is no need to check azureml prefix, as the path is checked for existence here.